Test/CoreSDK.Test/Shared/Extensibility/Implementation/ClockTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using Microsoft.ApplicationInsights.TestFramework;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class ClockTest
+    {
+        [TestMethod]
+        public void InstanceReturnsDefaultClockInstanceUsedByProductCode()
+        {
+            IClock instance = Clock.Instance;
+            Assert.IsType<Clock>(instance);
+        }
+
+        [TestMethod]
+        public void InstanceCanBeReplacedByTestsToControlClock()
+        {
+            TestableClock.Instance = new StubClock();
+            Assert.IsType<StubClock>(Clock.Instance);
+        }
+
+        [TestMethod]
+        public void InstanceCanBeSetToNullByTestsToRestoreDefaultClockImplementation()
+        {
+            TestableClock.Instance = null;
+            Assert.IsType<Clock>(Clock.Instance);
+        }
+
+        [TestMethod]
+        public void TimeReturnsHighPrecisionDateTimeOffsetValue()
+        {
+            IClock clock = Clock.Instance;
+
+            TimeSpan timestamp1 = clock.Time;
+            new ManualResetEvent(false).WaitOne(TimeSpan.FromTicks(50));
+            TimeSpan timestamp2 = clock.Time;
+
+            Assert.True(timestamp2 - timestamp1 > TimeSpan.FromTicks(5));
+            Assert.True(timestamp2 - timestamp1 < TimeSpan.FromTicks(5000));
```

Comment:
If you only waited 50 ticks then checking for less than 5000 seems a bit high...
This test doesn't seem to be particularly useful with such a big margin of error and given that it is timing dependent I don't see how you could make it precise enough to be useful.